### PR TITLE
Fix #1030: docker:build does not support 'network' option

### LINKED ISSUE #1030: docker:build does not support 'network' option
### Description

The command `docker build` supports the `--network=<name>` option, but the docker:build configuration in the plugin does not.

### Info

* d-m-p version : 0.23.0

* Maven version (`mvn -v`) :
```
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T11:41:47-05:00)
Maven home: /opt/apache-maven-3.3.9
Java version: 1.8.0_152, vendor: Oracle Corporation
Java home: /usr/java/jdk1.8.0_152/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.10.0-862.3.2.el7.x86_64", arch: "amd64", family: "unix"
```
* Docker version : Docker version 1.13.1, build 94f4240/1.13.1

* If it's a feature request, what is your use case :

There is an issue with the corporate VPN that I must use, where if you use the SSL version, everything works as expected; but if you use the IPSec version, the standard docker build process via the maven plugin fails because it apparently cannot receive responses from the remote host.  This despite multiple attempts at turning off  firewalls, adding the `iptables -P FORWARD ACCEPT` command, restarting docker, wiping out docker0 and restarting docker again, and again, and again, etc.

Doing the build manually, the same thing happens, until I add the `--network=host` option to the `docker build` command; then it works.

Now, the 'obvious' solution is to either (a) use the SSL version of the VPN, or (b) figure out what the network settings are needed to make it work with the IPSec version.  As to (a), I've been politely requested to not use that version of the VPN [for a really silly reason, but I can't make them fix it], and for (b) I'm not enough of a network guru to figure out what the correct settings are.

What I can do, is look at the docker build command's options, see the 'network' setting, and use it...

Except it isn't supported by docker-maven-plugin.

Please?

### PR DESCRIPTION
**Pull request by rohanKanojia**
Fix #1030

### CODE REVIEW

**Comment by rohanKanojia** [comment]
@albertschwarzkopf @fforloff: Hi, I've raised this PR to support this small feature. Could you guys please test it out for your use case and provide some feedback whenever you get some time?

You just need to pull this PR and build SNAPSHOT:
```
1. Clone repo: git clone <repo-url>
2. Pull my Branch: git fetch origin pull/1288/head:pullRequest1288 && git checkout pullRequest1288
3. Build SNAPSHOT : mvn clean install -DskipTests
4. Use that SNAPSHOT in your project
```

**Comment by rohanKanojia** [comment]
The network mode is right now added in build configuration like this:
```
        <configuration>
          <images>
            <image>
              <!-- Add your dockerhub user name here -->
              <name>rohankanojia/helloworld-java:${project.version}</name>
              <alias>hello-world</alias>
              <build>
                <from>openjdk:latest</from>
                <network>host</network>
                <assembly>
                  <descriptorRef>artifact</descriptorRef>
                </assembly>
                <cmd>java -jar maven/${project.name}-${project.version}.jar</cmd>
              </build>
              <run>
                <wait>
                  <log>Hello World!</log>
                </wait>
              </run>
            </image>
          </images>
        </configuration>

```

During testing, I could see networkmode query parameter being appended in verbose logs:
```
~/work/repos/jib-helloworld-sample : $ mvn docker:build -Ddocker.verbose=build,api
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------< io.fabric8.dmp.samples:dmp-sample-helloworld >------------
[INFO] Building dmp-sample-helloworld 0.31-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- docker-maven-plugin:0.31-SNAPSHOT:build (default-cli) @ dmp-sample-helloworld ---
[INFO] Copying files to /home/rohaan/work/repos/jib-helloworld-sample/target/docker/rohankanojia/helloworld-java/0.31-SNAPSHOT/build/maven
[INFO] Building tar: /home/rohaan/work/repos/jib-helloworld-sample/target/docker/rohankanojia/helloworld-java/0.31-SNAPSHOT/tmp/docker-build.tar
[INFO] DOCKER> [rohankanojia/helloworld-java:0.31-SNAPSHOT] "hello-world": Created docker-build.tar in 100 milliseconds
[INFO] DOCKER> POSTing to unix://127.0.0.1:1/v1.40/build?forcerm=1&networkmode=host&nocache=0&t=rohankanojia%2Fhelloworld-java%3A0.31-SNAPSHOT with contents of file /home/rohaan/work/repos/jib-helloworld-sample/target/docker/rohankanojia/helloworld-java/0.31-SNAPSHOT/tmp/docker-build.tar
[INFO] DOCKER> Step 1/3 : FROM openjdk:latest
[INFO] DOCKER> 
[INFO] DOCKER> ---> 8a8b42cf3239
[INFO] DOCKER> Step 2/3 : COPY maven /maven/
[INFO] DOCKER> 
[INFO] DOCKER> ---> Using cache
[INFO] DOCKER> ---> 28a3f0970578
[INFO] DOCKER> Step 3/3 : CMD java -jar maven/dmp-sample-helloworld-0.31-SNAPSHOT.jar
[INFO] DOCKER> 
[INFO] DOCKER> ---> Using cache
[INFO] DOCKER> ---> 0d396cbafb6d
[INFO] DOCKER> Successfully built 0d396cbafb6d
[INFO] DOCKER> Successfully tagged rohankanojia/helloworld-java:0.31-SNAPSHOT
[INFO] DOCKER> [rohankanojia/helloworld-java:0.31-SNAPSHOT] "hello-world": Built image sha256:0d396
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.607 s
[INFO] Finished at: 2019-11-09T20:06:14+05:30
[INFO] ------------------------------------------------------------------------
~/work/repos/jib-helloworld-sample : $ 

```

File: doc/changelog.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
   - Update to jnr-unixsocket 0.23
   - Add null check for null instance in config.json for email #1262
   - Allow merging of image configurations using <imagesMap> ([#360](https://github.com/fabric8io/docker-maven-plugin/issues/360))
+  - docker:build does not support 'network' option #1030
```
Comment (rhuss) [inline doc/changelog.md]: Could please formulate that in a positive way ? Like

```
- Add docker:build support for 'network' option #1030
```

**Comment by bellackn** [comment]
Hey guys, just a quick heads-up that this feature is still not documented. https://dmp.fabric8.io/#build-configuration 
However, it works as expected. Thanks!

**Comment by rohanKanojia** [comment]
@bellackn : Thanks for reporting this :+1: . Could you please raise a PR to add some documentation for this if you have time?

**Pull request merged**